rules/DowngradePhp80/NodeAnalyzer/UnnamedArgumentResolver.php

```diff
@@ -54,66 +106,62 @@ public function resolveFromReflection(
             }
         }
 
-        $setArgumentPositoins = array_keys($unnamedArgs);
-        $highestParameterPosition = max($setArgumentPositoins);
-        if (! is_int($highestParameterPosition)) {
-            throw new ShouldNotHappenException();
-        }
-
-        $unnamedArgs = $this->fillArgValues($highestParameterPosition, $unnamedArgs, $functionLikeReflection);
-        ksort($unnamedArgs);
-
         return $unnamedArgs;
     }
 
-    private function shouldSkipParam(Arg $arg, ParameterReflection $parameterReflection): bool
-    {
-        if (! $this->nodeNameResolver->isName($arg, $parameterReflection->getName())) {
-            return true;
-        }
-
-        return $this->areArgValueAndParameterDefaultValueEqual($parameterReflection, $arg);
-    }
-
-    private function areArgValueAndParameterDefaultValueEqual(ParameterReflection $parameterReflection, Arg $arg): bool
-    {
-        // arg value vs parameter default value
-        if ($parameterReflection->getDefaultValue() === null) {
-            return false;
-        }
-
-        $defaultValue = $this->defaultParameterValueResolver->resolveFromParameterReflection($parameterReflection);
-
-        // default value is set already, let's skip it
-        return $this->valueResolver->isValue($arg->value, $defaultValue);
-    }
-
     /**
      * @param Arg[] $unnamedArgs
+     * @param ParameterReflection[]|PhpParameterReflection[] $parameters
      * @return Arg[]
      */
-    private function fillArgValues(
-        int $highestParameterPosition,
+    private function fillFromJumpedNamedArgs(
```

Comment:
It seems there is quiote a lot going on in this service now.
Could you extract this logic to an external service?

Reply:
implemented https://github.com/rectorphp/rector-src/pull/466/commits/4c4b9308822c33c0aea8161516a8229510c16a46